### ISSUE TITLE
[7.0] IJW support in newer version of CMake breaks the build on empty TargetFramework

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#88806]
### Error Blob

```json
{
  "ErrorMessage": "",
  "BuildRetry": false,
  "ErrorPattern": "The TargetFramework value '' was not recognized\\..*ijw\\.vcxproj",
  "ExcludeConsoleLog": true
}
```

### Reproduction Steps

- PR: https://github.com/dotnet/runtime/pull/88758
- Queue: `Build windows x86 release MSBuild_CMake`
- Run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=337257&view=logs&j=16bd8e89-36f9-5867-227e-26683d5e0a13&t=4b75167b-259c-5825-6376-e1efaa150784&l=4693
- Full error:
```
##[error]src\native\corehost\corehost.proj(151,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command ""D:\a\_work\1\s\src\native\corehost\build.cmd" Release x86 apphostver 7.0.10 hostver 7.0.10 fxrver 7.0.10 policyver 7.0.10 commit de74b98a4cda34218c05e78c238b272f0a2790b5 rid win-x86 portable incremental-native-build rootdir D:\a\_work\1\s\ msbuild runtimeflavor CoreCLR runtimeconfiguration Release" exited with code 1.
```

And a few lines above:
```
  D:\a\_work\1\s\.dotnet\sdk\7.0.109\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(92,5): error NETSDK1013: The TargetFramework value '' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly. [D:\a\_work\1\s\artifacts\obj\win-x86.Release\corehost\ide\test\ijw\ijw.vcxproj]
```

From a conversation with @hoyosjs , seems related to cmake.

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=337257
**Error message validated:** `The TargetFramework value '' was not recognized\..*ijw\.vcxproj`
**Result validation: :white_check_mark:** Known issue matched with the provided build.
**Validation performed at:** 7/13/2023 12:45:27 AM UTC

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by carlossanlop** [comment]
Hit this in the branding PR for 7.0.11: https://github.com/dotnet/runtime/pull/89801

**Comment by jkoritzinsky** [comment]
Fixed in #88809
